Enable the @typescript-eslint/no-unused-vars eslint rule

tiago154 opened the pull request:
## Description:

### ✅ Summary of Changes

This PR enables the ESLint rule **`@typescript-eslint/no-unused-vars`** as requested in the issue and applies the necessary code adjustments across the project.

#### 🔧 What was done:
- Activated the rule `@typescript-eslint/no-unused-vars` in the ESLint config.
- Updated ~70 files to comply with the rule:
  - Replaced unused variables with a `_` prefix where appropriate.
  - Added inline ESLint disable comments (`eslint-disable-next-line`) for specific cases where the variable or code block seemed important for context, readability, or future use.
- Ensured no linting errors remain related to this rule.

---

### ❓ Clarification

Some cases were handled with inline disable comments instead of removing the variable entirely, to avoid accidental breaking changes or loss of intent.  
If a different approach is preferred (e.g., stricter removal or alternative handling), I’m happy to adjust the implementation accordingly — just let me know!

---

### 🙌 Next Steps

Please review and let me know if:
- Any file should be handled differently.
- You prefer removal instead of disabling in certain areas.
- Additional rules should be enforced or reverted.

I’m available to make any follow-up improvements needed.

---

### 🎃 Hacktoberfest Note

I'm participating in **Hacktoberfest**, so if this PR is accepted, please add the label:

`hacktoberfest-accepted`

Thank you!

#1784 

## Please complete the following:

- [x] I have added screenshots for all UI updates
- [x] I process any text displayed to the user through translateText() and I've added it to the en.json file
- [x] I have added relevant tests to the test directory
- [x] I confirm I have thoroughly tested these changes and take full responsibility for any bugs introduced

## Please put your Discord username so you can be contacted if a bug or regression is found:

DISCORD_USERNAME

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/openfrontio/OpenFrontIO?pullRequest=2130) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/openfrontio/OpenFrontIO?pullRequest=2130) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/openfrontio/OpenFrontIO?pullRequest=2130) it.</sub>

evanpelle commented:
> > Is there a way we can ignore unused parameters instead of _ prefix?
> 
> Thanks for the question! To avoid renaming unused parameters with a _ prefix, one possible configuration would be:
> 
> ```js
> "@typescript-eslint/no-unused-vars": [
>   "error",
>   {
>     args: "none",
>     argsIgnorePattern: "^_",
>     varsIgnorePattern: "^_",
>     caughtErrors: "none"
>   },
> ],
> ```
> 
> With this setup, unused parameters don’t trigger errors even without the _ prefix, while other unused variables are still checked. For example:
> 
> ✅ Unused parameters do not need _:
> 
> ```js
> function example(a, b) {
>   console.log(a);
>   // `b` is unused but no error
> }
> ```
> 
> ✅ catch parameters are also ignored:
> 
> ```js
> try {
>   doSomething();
> } catch (e) {
>   // not using `e` is fine
> }
> ```
> 
> ❌ Regular unused variables still need _ to be ignored:
> 
> ```js
> const unused = 123; // ❌ error
> const _ignored = 123; // ✅ ignored
> ```
> 
> If this direction makes sense for the project, I can update the PR accordingly — just let me know your preference before I change anything.

I would prefer to remove the check for arguments. and then let's remove the _ exception. so any var that is not an arg must be used.

tiago154 commented:
@evanpelle I've updated the lint configuration and adjusted unused variables. Some were ignored as they may be useful later, others were removed. Let me know if you think any further changes are needed.

tiago154 commented:
Hi @evanpelle 

I'm participating in Hacktoberfest and saw that my PR was merged — thank you!
Could you please add the **hacktoberfest-accepted** label so it counts toward the event?
Thanks a lot! 🙏
